src/control/cmd/daos_agent/infocache.go

```diff
@@ -60,17 +60,48 @@ func (aic *attachInfoCache) loadBalance(numaNode int) int {
 	return deviceIndex
 }
 
+// selectRemote is called when no local network adapters
+// was detected on the local NUMA node and we need to pick
+// one attached to a remote NUMA node. Some balancing is
+// still required here to avoid overloading a specific interface
+func (aic *attachInfoCache) selectRemote() (int, int) {
+	aic.mutex.Lock()
+
+	deviceIndex := invalidIndex
+	// restart from previous NUMA node
+	numaNode := aic.defaultNumaNode
+
+	for i := 0; i < len(aic.numaDeviceMarshResp); i++ {
+		node := (numaNode + i) % len(aic.numaDeviceMarshResp)
+		numDevs := len(aic.numaDeviceMarshResp[node])
+		if numDevs > 0 {
+			numaNode = node
+			deviceIndex = aic.currentNumaDevIdx[numaNode]
+			aic.currentNumaDevIdx[numaNode] = (deviceIndex + 1) % numDevs
+			break
+		}
+	}
+
+	// update the default with the one we finally picked up
+	aic.defaultNumaNode = numaNode
+
+	aic.mutex.Unlock()
```

Comment:
can just defer the unlock after taking it at the beginning of this function

Reply:
done